backend/s3/s3.go

```diff
@@ -4517,8 +4517,8 @@ func (f *Fs) Command(ctx context.Context, name string, arg []string, opt map[str
 				st.Status = "Not an S3 object"
 				return
 			}
-			if o.storageClass == nil || (*o.storageClass != "GLACIER" && *o.storageClass != "DEEP_ARCHIVE") {
-				st.Status = "Not GLACIER or DEEP_ARCHIVE storage class"
+			if o.storageClass != nil && (*o.storageClass != "GLACIER" && *o.storageClass != "DEEP_ARCHIVE" && *o.storageClass != "INTELLIGENT_TIERING") {
```

Comment:
This changes the meaning slightly if the storage class isn't set.

Before an unset storage class was considered to be not restorable, but with this change it will be restored.

I'm not sure how likely an unset storage class is - maybe this is academic - what do you think?

Reply:
Not academic; this would have changed semantics. I added a --no-traverse check.